auth0/src/main/java/com/auth0/android/request/internal/JwksDeserializer.java

```diff
@@ -0,0 +1,66 @@
+package com.auth0.android.request.internal;
+
+import android.util.Base64;
+import android.util.Log;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+class JwksDeserializer implements JsonDeserializer<Map<String, PublicKey>> {
+
+    private static final String RSA_ALGORITHM = "RS256";
+    private static final String USE_SIGNING = "sig";
+
+    @Override
+    public Map<String, PublicKey> deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        if (!json.isJsonObject() || json.isJsonNull() || json.getAsJsonObject().entrySet().isEmpty()) {
+            throw new JsonParseException("jwks json is not a valid json object");
```

Comment:
Is this the appropriate exception type to throw here? Since the issue is the json is either not a JsonObject, or is null, or is an empty object - not an error parsing the JSON, it just doesn't meet other requirements. Also perhaps worth including in the exception message information that the json needs to be a non-empty valid JSON object

Reply:
Yes, this is the same 3 line block used on the rest of the parsers. Parsers will only kick in if we explicitly tell the library that we want to parse a given response as that type. So, for this case that would only be for the endpoint that I added. 

If the JSON present on that response is other than a non-empty object, we fail with that exception that is [part of GSON](https://static.javadoc.io/com.google.code.gson/gson/2.6.2/com/google/gson/JsonParseException.html), symbolizing in a way that the JSON input was unexpected.

Reply:
Ok, then it makes sense to use the `JsonParseException`. I do think it may be worthwhile to include info in the message about it needing to be a non-empty json object, otherwise the message can be misleading (an empty object is a valid json object)